src/pykeen/models/unimodal/node_piece.py

```diff
@@ -0,0 +1,159 @@
+
+    This model uses the :class:`pykeen.nn.emb.NodePieceRepresentation` instead of a typical
+    :class:`pykeen.nn.emb.Embedding` to more efficiently store representations.
+    ---
+    citation:
+        author: Galkin
+        year: 2021
+        link: https://arxiv.org/abs/2106.12144
+        github: https://github.com/migalkin/NodePiece
+    """
+
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        embedding_dim=DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE,
+    )
+
+    def __init__(
+        self,
+        *,
+        triples_factory: CoreTriplesFactory,
+        num_tokens: int = 2,
+        embedding_dim: int = 64,
+        embedding_specification: Optional[EmbeddingSpecification] = None,
+        interaction: HintOrType[Interaction] = DistMultInteraction,
+        aggregation: Hint[Callable[[torch.Tensor, int], torch.Tensor]] = None,
+        shape: Optional[Sequence[int]] = None,
+        **kwargs,
+    ) -> None:
+        """
+        Initialize the model.
+
+        :param triples_factory:
+            the triples factory used for tokenization
+        :param embedding_dim:
+            the embedding dimension. Only used if embedding_specification is not given.
+        :param embedding_specification:
+            the embedding specification.
+        :param interaction:
+            the interaction module, or a hint for it.
+        :param aggregation:
+            aggregation of multiple token representations to a single entity representation. By default,
+            this uses :func:`torch.mean`. It could also use other aggregations like :func:`torch.sum`,
+            :func:`torch.max`, or even trainable aggregations e.g., ``MLP(mean(MLP(tokens)))``
+            (cf. DeepSets from [zaheer2017]_) if given value ``"mlp"``.
+
+            We could also have aggregations which result in differently shapes output, e.g. a concatenation of all
+            token embeddings resulting in shape ``(k * d,)``. In this case, `shape` must be provided.
+
+            The aggregation takes two arguments: the (batched) tensor of token representations, in shape
+            ``(*, k, *dt)``, and the index along which to aggregate.
+        :param shape:
+            the shape of an individual representation. Only necessary, if aggregation results in a change of dimensions.
+            this will only be necessary if the aggregation is an *ad hoc* function.
+        :param kwargs:
+            additional keyword-based arguments passed to :meth:`ERModel.__init__`
+        """
+        if not triples_factory.create_inverse_triples:
+            logger.warning(
+                "The provided triples factory does not create inverse triples. However, for the node piece"
+                "representations inverse relation representations are required. Thus, the implicitly created inverse "
+                "relations are only trained via the NodePiece mechanism, but not as part of the 'normal' training.",
+            )
```

Comment:
@cthoyt I would also support to raise an exception here (since the models here are just a convenient way to compose the more general `ERModel` according to some papers' configurations).

Reply:
Yes I think raising an exception rather than trying to compensate in a complicated way is a good idea

Reply:
ac97a56